core/src/test/scala/kafka/tools/LogCompactionTester.scala

```diff
@@ -153,12 +154,13 @@ object LogCompactionTester {
     try {
       val topicConfigs = Map(TopicConfig.CLEANUP_POLICY_CONFIG -> TopicConfig.CLEANUP_POLICY_COMPACT)
       val newTopics = topics.map(name => new NewTopic(name, 1, 1).configs(topicConfigs.asJava)).asJava
-      adminClient.createTopics(newTopics).all.get()
+      adminClient.createTopics(newTopics).all.get
 
+      var allTopics: Seq[String] = Seq()
       TestUtils.waitUntilTrue(() => {
-        val newTopics = adminClient.listTopics.names.get()
-        topics.forall(topicName => newTopics.contains(topicName))
-      }, "timed out waiting for topics")
+        allTopics = adminClient.listTopics.names.get.asScala.toSeq
+        topics.forall(topicName => allTopics.contains(topicName))
```

Comment:
Seems like you can do the `filter` here instead of `forall` and then the message creation becomes simpler.